lib/PuppeteerSharp/Launcher.cs

```diff
@@ -169,6 +169,28 @@ public Launcher()
 
         }
         
+        /// <summary>
+        /// Connects to an existing browser.
+        /// </summary>
+        /// <param name="options">Options for connecting.</param>
+        /// <returns>A connected browser.</returns>
+        public async Task<Browser> ConnectAsync(ConnectOptions options)
+        {
+            try
+            {
+                var connectionDelay = options.SlowMo;
+                var keepAliveInterval = options.KeepAliveInterval;
+
+                _connection = await Connection.Create(options.BrowserWSEndpoint, connectionDelay, keepAliveInterval);
+
+                return await Browser.CreateAsync(_connection, options, () => Task.CompletedTask);
```

Comment:
Puppeteer is calling Browser.Close here https://github.com/GoogleChrome/puppeteer/blob/v1.0.0/lib/Launcher.js#L200

Reply:
If we're going for exactly the same, I'll change it. But to me, it doesn't make sense that disposing a browser that didn't launch the executable get it to close, especially since "Disposing" of the browser will always call the browser.close and by convention, Disposable classes should be in `using` blocks?